### ISSUE TITLE
Publish the precompiled wheels for python 3.11

### ISSUE THREAD

**Issue by martinResearch** [nerfstudio-project/gsplat#462]
Thank you for this library. It would be great to publish the pre-compiled  python wheels for python 3.11, could that be added to the release script?

**Comment by martinResearch** [comment]
Hi, any blocker on this I can help wit ? This prevents me from switching to python 3.11 on a large project, which is not ideal.

**Comment by liruilong940607** [comment]
hey srry for the late reply. i think the question is where to host it. I don't want to host all python versions in github release ...

**Comment by martinResearch** [comment]
I'm not sure what a great alternative to GitHub Releases would be for distributing the binaries. I listed a few potential options for publishing to PyPI in issue #331 , but none of them seemed ideal given the size constraints and compatibility concerns.

If the main issue is the total size of the release, one option could be to drop support for CUDA 11.8 to reduce the number of build variants.

You also mentioned in that thread that "gsplat is fairly lightly dependent on PyTorch, so a precompiled wheel using a specific PyTorch version might just work across a wide range of versions." It could be worth testing whether a wheel built for the latest PyTorch version is also compatible with older ones. Maybe we can decouple further gsplat from the pytorch version to the point where we can drop the pytorch version suffix from the wheel name?

**Comment by liruilong940607** [comment]
> Maybe we can decouple further gsplat from the pytorch version to the point where we can drop the pytorch version suffix from the wheel name?

Yeah i like the idea of exploring this path. And actually I think maybe even possible to decouple the python version suffix as well because I can't thinking of anything that depends on specific python versions. (something like --manylinux-- but I'm not familiar with how to set this up)

**Comment by liruilong940607** [comment]
I think https://github.com/pypa/cibuildwheel this tool could be used to build multi-python-verison wheel.

**Comment by liruilong940607** [comment]
Ok i looked into it but wasn't able to make it work with cibuildwheel ...

**Comment by martinResearch** [comment]
this could be of interest once available https://discuss.python.org/t/wheelnext-wheel-variants-an-update-and-a-request-for-feedback/102383

**Comment by martinResearch** [comment]
I compiled gsplat for 20 different configurations with pytorch 2.6 in a fork here https://github.com/martinResearch/gsplat/releases/tag/1.5.3

| OS | Python | CUDA | Wheel 
|----|--------|------|-------|
| Linux | 3.10 | cu118 | `gsplat-1.5.3+pt26cu118-cp310-cp310-linux_x86_64.whl` | 
| Linux | 3.11 | cu118 | `gsplat-1.5.3+pt26cu118-cp311-cp311-linux_x86_64.whl` | 
| Linux | 3.12 | cu118 | `gsplat-1.5.3+pt26cu118-cp312-cp312-linux_x86_64.whl` | 
| Linux | 3.13 | cu118 | `gsplat-1.5.3+pt26cu118-cp313-cp313-linux_x86_64.whl` | 
| Linux | 3.10 | cu124 | `gsplat-1.5.3+pt26cu124-cp310-cp310-linux_x86_64.whl` | 
| Linux | 3.11 | cu124 | `gsplat-1.5.3+pt26cu124-cp311-cp311-linux_x86_64.whl` |
| Linux | 3.12 | cu124 | `gsplat-1.5.3+pt26cu124-cp312-cp312-linux_x86_64.whl` | 
| Linux | 3.13 | cu124 | `gsplat-1.5.3+pt26cu124-cp313-cp313-linux_x86_64.whl` | 
| Linux | 3.10 | cu126 | `gsplat-1.5.3+pt26cu126-cp310-cp310-linux_x86_64.whl` | 
| Linux | 3.11 | cu126 | `gsplat-1.5.3+pt26cu126-cp311-cp311-linux_x86_64.whl` | 
| Linux | 3.12 | cu126 | `gsplat-1.5.3+pt26cu126-cp312-cp312-linux_x86_64.whl` | 
| Linux | 3.13 | cu126 | `gsplat-1.5.3+pt26cu126-cp313-cp313-linux_x86_64.whl` | 
| Windows | 3.10 | cu124 | `gsplat-1.5.3+pt26cu124-cp310-cp310-win_amd64.whl` | 
| Windows | 3.11 | cu124 | `gsplat-1.5.3+pt26cu124-cp311-cp311-win_amd64.whl` |
| Windows | 3.12 | cu124 | `gsplat-1.5.3+pt26cu124-cp312-cp312-win_amd64.whl` | 
| Windows | 3.13 | cu124 | `gsplat-1.5.3+pt26cu124-cp313-cp313-win_amd64.whl` | 
| Windows | 3.10 | cu126 | `gsplat-1.5.3+pt26cu126-cp310-cp310-win_amd64.whl` | 
| Windows | 3.11 | cu126 | `gsplat-1.5.3+pt26cu126-cp311-cp311-win_amd64.whl` | 
| Windows | 3.12 | cu126 | `gsplat-1.5.3+pt26cu126-cp312-cp312-win_amd64.whl` |
| Windows | 3.13 | cu126 | `gsplat-1.5.3+pt26cu126-cp313-cp313-win_amd64.whl` | 

They can be installed with pip using 
* For PyTorch 2.6 + CUDA 12.4:
 ```pip install gsplat==1.5.3 --find-links https://martinresearch.github.io/gsplat/whl/pt26cu124/```

* For PyTorch 2.6 + CUDA 12.6:
 ```pip install gsplat==1.5.3 --find-links https://martinresearch.github.io/gsplat/whl/pt26cu126/```

* For PyTorch 2.6 + CUDA 11.8 (Linux only):
 ```pip install gsplat==1.5.3 --find-links https://martinresearch.github.io/gsplat/whl/pt26cu118/```


By reusing intermediate obj files in the compilation I managed to  reduce the compilation cost on CI by 3.5×. 

I explored ways to reduce the number of wheels by generating some wheels that can support ranges of python version and pytorch version, but unfortunately I did not get anything working. 

@liruilong940607 happy to submit a PR with my current code if you think that it is a good idea.

**Comment by martinResearch** [comment]
I modified my CI job and made new release in my fork using the latest pytorch version for each python and cuda version instead of using a fixed pytorch version, so that we do not favor consumers using older version of pytorch over people using the latest version.
The wheel are here https://github.com/martinResearch/gsplat/releases/tag/1.5.3
and accessible to  pip here https://martinresearch.github.io/gsplat/whl

This creates 28 wheels. Each wheel is about 20 MB and the total release size us about 560 MB. 
To keep build times manageable, compiled CUDA objects are reused across Python versions, so just 7 CI build jobs are enough to produce all 28 wheels, using only about 3× the compute of the original single-Python setup.

@liruilong940607  happy to submit a PR.

**Comment by LihanChen2004** [comment]
For what it's worth (related to the "precompiled wheels" ask here): I forked the repo and built a fresh batch of precompiled wheels from a recent upstream commit, in case it helps anyone who wants to `pip install` without compiling from source.

- Base: [nerfstudio-project/gsplat@2b902ff](https://github.com/nerfstudio-project/gsplat/commit/2b902ff1891fc7f73f0f9b8c8bfc932cef2b198c)
- My build (unofficial, version `1.5.4`): https://github.com/LihanChen2004/gsplat/releases/tag/v1.5.4

Wheels are Linux (Ubuntu 24.04), Python 3.11/3.12, for PyTorch 2.9.1 / 2.10.0 / 2.11.0 / 2.13.0 across CUDA 12.8 / 13.0 (PyTorch 2.13.0 builds CUDA 13.0 only). Not an official release — just sharing in case it saves someone a compile.
